File: packages/pensopay_android/android/src/main/kotlin/com/flutter/pensopay/networking/pensopayapi/pensopaylink/models/PPPayment.kt
package com.flutter.pensopay.networking.pensopayapi.pensopaylink.models

import android.src.main.kotlin.com.flutter.pensopay.networking.pensopayapi.pensopaylink.models.PPOrder
import org.json.JSONObject

class PPPayment: JSONObject() {

    // Properties
    val id: Int = 0
    val order_id: String = ""
    val type: String = ""
    val amount = 0
    val captured = 0
    val refunded = 0
    val currency: String = ""
    val state: String = ""
    val facilitator: String = ""
    val reference: String = ""
    val testmode: Boolean = false
    val autocapture: Boolean = false
    val link: String = ""
    val callback_url: String = ""
    val success_url: String = ""
    val cancel_url: String = ""
    val order: List<PPOrder>? = null
    val variables: List<Any>? = null
    val expires_at: String = ""
    val created_at: String = ""
    val updated_at: String = ""
}

File: packages/pensopay_ios/ios/Classes/Networking/Pensopay API/Pensopay Link/Models/PPPayment.swift
//
//  PPPayment.swift
//  PensopaySDK
//
//  Created on 07/26/2022
//  Copyright © 2022 Pensopay. All rights reserved.
//

import Foundation

public class PPPayment: Codable {
    
    // MARK: - Properties

    public var id: Int
    public var order_id: String
    public var type: String
    public var amount: Int
    public var captured: Int
    public var refunded: Int
    public var currency: String
    public var state: String
    public var facilitator: String

    //public var reference: String {
    //    get {
    //        return reference
    //    }
    //    set (value) {
    //        reference = String(value ?? "")
    //    }
    //}

    public var testmode: Bool
    public var autocapture: Bool
    public var link: String
    public var callback_url: String?
    public var success_url: String?
    public var cancel_url: String?
    public var order: Array<String>?
    public var variables: Array<String>?
    public var expires_at: String
    public var created_at: String
    public var updated_at: String
}